webapp-django/crashstats/signature/static/signature/js/signature_panel.js

```diff
@@ -0,0 +1,33 @@
+SignatureReport.Panel = function (panelName, onDelete) {
+
+    // For accessing this from inside functions.
+    var that = this;
+
+    // Make the HTML elements.
+    this.$panelElement = $('<div>', {'class': 'panel'});
+    var $headerElement = $('<div>', {'class': 'title'});
+    var $headingElement = $('<h2>', {
+        'text': SignatureReport.capitaliseHeading(panelName)
+    });
+    var $deleteButton = $('<a>', {
+        'href': '#',
+        'class': 'options delete',
+        'text': 'X'
+    });
+    var $bodyElement = $('<div>', {'class': 'body'});
+    this.$contentElement = $('<div>', {'class': 'content'});
+
+    // Bind the delete function to the delet button.
+    $deleteButton.on('click', function (e) {
+        e.preventDefault();
+        that.$panelElement.remove();
+        onDelete();
+    });
+
+    // Append everything.
+    $headerElement.append($headingElement, $deleteButton);
+    SignatureReport.addLoaderToElement(this.$contentElement);
+    $bodyElement.append(this.$contentElement);
+    this.$panelElement.append($headerElement, $bodyElement);
```

Comment:
Is this clearer?

    this.$panelElement.append(
        $headerElement.append(
            $headingElement,
            $deleteButton
        ),
        $bodyElement.append(
            this.$contentElement
        )
    );